test/e2e_env/universal/healthcheck/policy.go

```diff
@@ -161,5 +184,33 @@ conf:
 			// empty response with EXIT_CODE = 0, or  'Ncat: Connection reset by peer.' with EXIT_CODE = 1
 			Expect(stdout).To(Or(BeEmpty(), ContainSubstring("Ncat: Connection reset by peer.")))
 		})
+
+		It("should mark host as unhealthy if it doesn't reply on health checks when Permissive mTLS enabled", func() {
```

Comment:
It sounds like those changes have 0 in common with the existing test (separate mesh, separate client&server etc.).
Can we move it to another file or at least a separate `Context`